### ISSUE TITLE
[Bug]: Fetch requests do not go through service worker

### ISSUE THREAD

**Issue by manuelei** [serwist/serwist#98]
### Provide environment information

"@serwist/next": "^8.4.4",
"@serwist/precaching": "^8.4.3",
"@serwist/sw": "^8.4.3",

### Which project is this issue for?

@serwist/next

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code help us address them faster

h

### To reproduce

page.tsx:
fetch('/local_test');

sw.ts:
self.addEventListener('fetch', event => {
  console.log('Fetching:', event.request.url); // Nothing shown in console (Chromium 121)
}

### Describe the bug

Fetch request is not going through service worker (the request does not show up in the console). I also tried 'navigationPreload: false', which did not help. I set the path in public/manifest.json correctly, I think: "start_url": "/PATH_AT_WHICH_PAGE.JSX_IS_SHOWN".

### Expected behavior

Fetch request is going through service worker, i.e. the request should be printed in the console.

### Additional information (if relevant)

I am trying to create a mock API, therefore using a local URL.

I actually think there are two bugs: I get an error in the console: 'GET https://example.com/local_test' ... net:ERR_FAILED: 1) The URL shown is from an old version of the service worker. It looks like it did not reload, despite 'skipWaiting: true, clientsClaim: true,' 2) I suspect there is an internal serwist fetch listener which is called first, returns a failing promise and then my custom fetch listener is not activated? If so, how can I serve the fetch request anyway?

**Comment by DuCanhGH** [comment]
1) The URL shown is from an old version of the service worker. It looks like it did not reload, despite 'skipWaiting: true, clientsClaim: true,'

The service worker does take some time to install. After that, you need to reload the page so that it is loaded under the control of the service worker. You may find having it reload automatically useful.

> 2) I suspect there is an internal serwist fetch listener which is called first, returns a failing promise and then my custom fetch listener is not activated? If so, how can I serve the fetch request anyway?

Please add your own `fetch` listeners before Serwist's code. Its listener calls `event.respondWith`, which ends the `fetch` event listener call chain, resulting in your code not being called. Thanks!

**Comment by manuelei** [comment]
Thanks for the reply. If I add my event listeners before installSerwist(), they are indeed called. Concerning the page reload, I though that skipWaiting should take care of that? https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope/skipWaiting

**Comment by DuCanhGH** [comment]
@manuelei `skipWaiting` and `clients.claim()` do kick out the current active worker, make the waiting worker active, and claim all clients, including this one, making every new request go through the newly active worker, but they are time-sensitive. So I mentioned reloading just to be sure, but you may also check when it goes active; at that point, there shouldn't be any outdated service worker.

Do note that I don't really understand "1) The URL shown is from an old version of the service worker. It looks like it did not reload, despite 'skipWaiting: true, clientsClaim: true,'" that well though. Sorry about that, but I think I need an illustration of the issue.

**Comment by manuelei** [comment]
Sorry, I noticed that the service worker actually updates now that I add my listeners before installSerwist(). But the code for the page (page.jsx) does not update...
And I now figured out why: I was seeing so many "NetworkFirst" console messages that I assumed that would be the default strategy for all documents...which it is not. After adjusting the 'runtimeCaching' property accordingly, my problems seem to be solved now. Thank you.
